build.gradle

```diff
@@ -5,12 +5,12 @@ buildscript {
         google()
         mavenCentral()
         maven { url "https://jitpack.io" }
-
     }
     dependencies {
+        // Upgrading AGP to a newer version also updates the data binding library.
```

Comment:
I'm not sure this is correct. It builds okay since the pre-processor plugin but the resulting binary doesn't work correctly on some handsets.

Reply:
I believe it is this issue: https://github.com/NightscoutFoundation/xDrip/issues/1796

As far as I know the problem is with the later databinding versions and not anything to do with mixed package names. I believe it only affects specific handsets or perhaps configurations. 

In regards to the tests inside android studio, its only the test filtering which doesn't work correctly if the test package name has mixed case? eg, running all tests works fine. I think this could be quite easy to resolve simply by adjusting the package names (on the first line of the test class) to lower case in the tests only.

Reply:
> In regards to the tests inside android studio, its only the test filtering which doesn't work correctly if the test package name has mixed case? eg, running all tests works fine. I think this could be quite easy to resolve simply by adjusting the package names (on the first line of the test class) to lower case in the tests only.

No, executing all tests fails as well. Renaming just the test package is not possible because test classes are tightly bound to there target classes and their package names must be identical. Otherwise, testing would not be possible the tests have to be completely rewritten and target classes have to be imported individually. Just try to rename the test package. The refactoring preview will show you hundreds of affected files.

Reply:
I'll do the tests and report in #2005. Let us focus on the changes in this PR.